src/collection/linked-node.ts

```diff
@@ -1,8 +1,8 @@
 interface LinkedNode {
-  prev: LinkedNode | null;
-  next: LinkedNode | null;
+  prev?: LinkedNode | null;
+  next?: LinkedNode | null;
 
   length(): number;
 }
 
-export default LinkedNode;
+export type { LinkedNode };
```

Comment:
Why we don't make it a default export anymore?

Reply:
It was changed due to [verbatimModuleSyntax](https://devblogs.microsoft.com/typescript/announcing-typescript-5-0/#verbatimmodulesyntax)  in ts 5.0. 

Since `LinkedNode` is a pure type, and ES modules don't have "default type" exports (i.e. `export default type SomeType`), we can only use named export.

I think a module with only default export to be a type is effectively an empty module (no runtime code) . If we add run time default export (`export default SomeValue` + `export default type SomeType`) they would conflict with each other.

Reply:
Good to know! I think we should change it to `export type { LinkedNode as default };` to keep it consistent with other parts of the app that the default export should somehow match the filename.